src/rules/no-declare.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  AST_NODE_TYPES,
+  ESLintUtils,
+} from "@typescript-eslint/experimental-utils";
+
+const createEslintRule = ESLintUtils.RuleCreator((ruleName) => ruleName);
+
+interface IOptions {
+  onlyIdentifiers?: string[];
+  excludeIdentifiers?: string[];
+}
+
+export default createEslintRule<IOptions[], string>({
+  name: "no-declare",
+  meta: {
+    messages: {
+      declareFound: "Using 'declare' is not allowed.",
+      declareIdentifierFound:
+        "Using 'declare' is not allowed for this identifier: '{{ name }}'.",
+    },
+    docs: {
+      description: "Forbid using 'declare' keyword.",
+      recommended: "warn",
+    },
+    type: "suggestion",
+    schema: [
+      {
+        type: "object",
+        properties: {
+          onlyIdentifiers: {
+            type: "array",
+            uniqueItems: true,
+            items: {
+              type: "string",
+            },
+          },
+        },
+      },
+      {
+        type: "object",
+        properties: {
+          excludeIdentifiers: {
+            type: "array",
+            uniqueItems: true,
+            items: {
+              type: "string",
+            },
+          },
+        },
+      },
+    ],
+  },
+  defaultOptions: [
+    {
+      onlyIdentifiers: [],
+      excludeIdentifiers: [],
+    },
+  ],
+  create(context, [options]) {
+    const { onlyIdentifiers, excludeIdentifiers } = options;
+    const onlyIdentifiersEnabled =
+      onlyIdentifiers && onlyIdentifiers?.length > 0;
```

Comment:
```suggestion
      !!onlyIdentifiers?.length;
```

Reply:
well guess what:
`Operator '>' cannot be applied to types 'boolean' and 'number'.`
also:
`Object is possibly 'undefined'.`